config/util.go

```diff
@@ -15,6 +15,39 @@ import (
 	"github.com/choria-io/go-choria/confkey"
 )
 
+// ProjectConfigurationFiles returns any configuration file in the specified directory and their parents directories.
+func ProjectConfigurationFiles(path string, err error) []string {
+	// TODO: Check if it works on Windows
+
+	var res []string
+
+	if !filepath.IsAbs(path) {
+		path, err = filepath.Abs(path)
+	}
+
+	var parent = filepath.Dir(path)
+	if parent != path {
+		res = ProjectConfigurationFiles(parent, err)
+	}
+
+	var config = filepath.Join(path, "choria.conf")
+
+	if FileExist(config) {
+		res = append(res, config)
+	}
+
+	return res
+}
+
+// FileExist checks if a file exist on disk
+func FileExist(path string) bool {
```

Comment:
I copied this function from `choria/util.go` because attempting to use it directly produce circular dependencies.  I am unsure however about if I should remove it from that other location and update the rest of the project to call `config.FileExist()`?

Reply:
its ok to duplicate, bloody go annoyances. the one in choria is probably used elsewhere so its fine